Lesson_22/js/selector.js

```diff
@@ -0,0 +1,78 @@
+export class Selector {
+   #prevClick = null;
+   #target = null;
+
+   constructor(list) {
+      this._list = list;
+      this._items = [...this._list.children];
+      list.onclick = this.onClick.bind(this);
+      list.onmousedown = e => e.preventDefault();
+      list.addEventListener('onChangeList', this.#handleChangeList.bind(this));
+   }
+
+   onClick(e) {
+      if (e.target.tagName !== "LI") return;
+
+      this.#target = e.target;
+
+      if (e.ctrlKey || e.metaKey) {
+         this.toggleSelect();
+      } else if (e.shiftKey) {
+         this.multipleSelect();
+      } else {
+         this.singleSelect();
+      }
+
+      this.#prevClick = e.target;
+   }
+
+   toggleSelect() {
+      this.#target.classList.toggle('selected');
+   }
+
+   singleSelect() {
+      this.#clearSelected();
+      this.#addSelected(this.#target);
+   }
+
+   multipleSelect() {
+      const [targetIndex, prevClickIndex] = this.#getIndex();
+      
+      if (!this.#prevClick) {
+         for (let i = 0; i <= targetIndex; i++) {
+            this.#addSelected(this._items[i]);
+         }
+      } else if (targetIndex > prevClickIndex) {
+         for (let i = prevClickIndex; i <= targetIndex; i++) {
+            this.#addSelected(this._items[i]);
+         }
+      } else if (prevClickIndex > targetIndex) {
+         for (let i = prevClickIndex; i >= targetIndex; i--) {
+            this.#addSelected(this._items[i]);
+         }
+      }
+   }
```

Comment:
this function can be shortened by at least 2 times. Try to redo it

Reply:
reduced